ideascube/serveradmin/catalog.py

```diff
@@ -646,28 +678,50 @@ def upgrade_packages(self, ids):
 
             self._catalog['installed'][ipkg.id] = (
                 self._catalog['available'][upkg.id])
-            self._persist_cache()
+            self._persist_catalog()
 
         for handler in used_handlers.values():
             handler.commit()
 
     # -- Manage local cache ---------------------------------------------------
-    def _load_cache(self):
-        if os.path.exists(self._cache_catalog):
-            with open(self._cache_catalog, 'r') as f:
-                self._catalog = yaml.safe_load(f.read())
+    def _load_catalog(self):
+        self._catalog = {'available': {}, 'installed': {}}
+
+        try:
+            catalog = load_from_file(self._catalog_cache)
+
+        except FileNotFoundError:
+            # That's ok
+            pass
+
+        else:
+            if catalog is not None:
+                if 'available' in catalog and 'installed' in catalog:
+                    # The cache on file is in the old format
+                    # https://github.com/ideascube/ideascube/issues/376
+                    self._catalog = catalog
+
+                else:
+                    self._catalog['available'] = catalog
+
+        try:
+            installed = load_from_file(self._installed_storage)
+
+        except FileNotFoundError:
+            # That's ok
+            pass
+
+        else:
+            if installed is not None:
+                self._catalog['installed'] = installed
 
-        # yaml.safe_load returns None for an empty file.
-        if not getattr(self, '_catalog', None):
-            self._catalog = {'installed': {}, 'available': {}}
-            self._persist_cache()
+        self._persist_catalog()
 
         self._package_caches = [self._local_package_cache]
-        os.makedirs(self._local_package_cache, exist_ok=True)
 
-    def _persist_cache(self):
-        with open(self._cache_catalog, 'w') as f:
-            f.write(yaml.safe_dump(self._catalog, default_flow_style=False))
+    def _persist_catalog(self):
+        persist_to_file(self._catalog_cache, self._catalog['available'])
```

Comment:
>> Keeping them as is has the merit of making for a much simpler diff to review. :stuck_out_tongue_winking_eye:

> Meeeh! ;)

I agree that's not a very strong argument. :wink:

>> What would separating them achieve?

> Something useless being removed? :)

I don't understand. Right now we have one attribute, you want to split it in two. That's not removing anything, that's adding something.